### DISCUSSION TITLE
Fetching config from AWS secrets store

### DISCUSSION THREAD

**Discussion by bullett445** [node-config/node-config#736] [Ideas]
Hi,
I would like to extend the package in a way that I can define a config fetching from AWS secrets store in the same way as the custom-environment-variables.json file works. 
aws-secrets.json5:
`{
      MyModule: {
           SecretApiKey: "arn:aws:secretsmanager:<Region>:<AccountId>:secret:SecretName-6RandomCharacters"
      }
}`

is it possible to add this? Has anyone done that before?

Jan

**Comment by jdmarshall** [comment]
Two ways I’ve solved this. 

Option A. Build your own facade around node-config which can pull in other values and expose your own get method (I recommend using lodash under the hood so you can use dotted names). The facade solution also facilitates updating values for expiring tokens, or reloadable config. For instance long-polling Consul for overrides. 

The other answer is setModuleDefaults, which has had a handful of weird bugs fixed in it in 4.0.1 and 4.1.

**Reply by jdmarshall** [reply]
In 4.3 or later I'd recommend the setModuleDefaults solution, and see `scan()` in util.js

**Comment by markstos** [comment]
One solution: Map all the values in Secrets to ENV vars which get loaded first, then load use custom-environment-variables.json to map those ENV vars into node-config.
